### PR TITLE
testall and friends updates

### PR DESCRIPTION
**Pull request by pasinskim**
Right now all the fields form test_meta bundle are parsed and added to the test results xml file. This file is then accessible while Jenkins is done with building and new test related fields are included in QA dashboard.

### CODE REVIEW

**Comment by pasinskim** [comment]
@kacfengine what do you think about this one?

File: tests/acceptance/testall [hunk 1]
```diff
@@ -389,6 +389,18 @@ export CFENGINE_TEST_OVERRIDE_WORKDIR TEMP CFENGINE_TEST_OVERRIDE_EXTENSION_LIBR
...
 
+    # Try to collect test metadata if any
+    if egrep -e "R: test description: " $OUTFILE > /dev/null; then
+      TEST_DESCRIPTION="$(egrep -e "R: test description" $OUTFILE | sed -e "s,.*test description: \(\w*\),\1,")"
+    fi
+    if egrep -e "R: test story_id: " $OUTFILE > /dev/null; then
+      TEST_STORY="$(egrep -e "R: test story_id" $OUTFILE | sed -e "s,.*test story_id: \([0-9][0-9]*\),\1,")"
+    fi
+    if egrep -e "R: test covers: " $OUTFILE > /dev/null; then
+      TEST_COVERS="$(egrep -e "R: test covers" $OUTFILE | sed -e "s,.*test covers:[ \t][ \t]*\(\w*\),\1,")"
```
Comment (kacfengine) [inline tests/acceptance/testall]: ``\w`` is not in POSIX. I'm pretty sure the non-Linux sed tools won't understand it.

The same with ``egrep -e``. Just use ``egrep`` without the ``-e``.

Comment (kacfengine) [inline tests/acceptance/testall]: And ``\t`` as well, not in POSIX.

**Comment by kacfengine** [comment]
Looks clean and nice though!

**Comment by pasinskim** [comment]
@kacfengine all POSIX incompatibilities fixed now. Also removed some 'egrep -e' stuff from other places.

File: tests/acceptance/testall [hunk 2]
```diff
@@ -402,40 +414,40 @@ export CFENGINE_TEST_OVERRIDE_WORKDIR TEMP CFENGINE_TEST_OVERRIDE_EXTENSION_LIBR
       # Some states are output by dcs.cf.sub, therefore check for both TEST
       # prefix and dcs.cf.sub prefix.
       ESCAPED_TEST="$(echo "($TEST|dcs.cf.sub)" | sed -e 's/\./\\./g')"
-      if egrep -e "R: .*$ESCAPED_TEST [XS]FAIL" $OUTFILE > /dev/null; then
+      if egrep "R: .*$ESCAPED_TEST [XS]FAIL" $OUTFILE > /dev/null; then
```
Comment (kacfengine) [inline tests/acceptance/testall]: Ah, so ``-e`` was already there, and so must have worked. Maybe I remembered that wrong, or only pure "grep" doesn't support it. Either way, removing it also works!

**Comment by kacfengine** [comment]
trigger build

**Comment by kacfengine** [comment]
@pasinskim: Should this also be cherry-picked?

**Pull request merged**